src/main/java/org/magicdgs/popgenlib/diversity/NucleotideDiversity.java

```diff
@@ -0,0 +1,160 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.magicdgs.popgenlib.diversity;
+
+import org.magicdgs.popgenlib.utils.FrequencyUtils;
+import org.magicdgs.popgenlib.utils.Verify;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.special.Gamma;
+import org.apache.commons.math3.util.FastMath;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Static methods to compute nucleotide diversity statistics.
+ *
+ * <p>REFERENCES:
+ * <ul>
+ *
+ * <li><a href="http://www.genetics.org/content/123/3/585">
+ * Tajima (1989): Statistical method for testing the neutral mutation hypothesis by DNA
+ * polymorphism, Genetics 123(3).
+ * </a></li>
+ *
+ * <li><a href="http://www.sciencedirect.com/science/article/pii/0040580975900209">
+ * Watterson (1975): On the number of segregating sites in genetical models without recombination,
+ * Theor. Popul. Biol. 7(2).
+ * </a></li>
+ *
+ * </ul>
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public final class NucleotideDiversity {
+
+    // cannot be instantiated
+    private NucleotideDiversity() {}
+
+    /**
+     * Computes Tajima's &pi; for a single site using allele frequencies.
+     *
+     * <p>Corresponds to formula 12 in
+     * <a href="http://www.genetics.org/content/123/3/585">Tajima (1989)</a>
+     *
+     * @param numberOfSamples   number of samples used for get the frequencies.
+     * @param alleleFrequencies frequencies for each allele.
+     */
+    public static double tajimasPi(final int numberOfSamples,
+            final List<Double> alleleFrequencies) {
+        FrequencyUtils.validateFrequencies(alleleFrequencies);
+        Verify.validate(numberOfSamples >= 2,
+                () -> "numberOfSamples should be at least 2 for computing Tajima's Pi: "
+                        + numberOfSamples);
+
+        // computes the sum of p2
+        final double pSquareSum =
+                alleleFrequencies.stream().mapToDouble(i -> FastMath.pow(i, 2)).sum();
+        // tajima's pi computed as in formula 12 in Tajima 1989
+        return (numberOfSamples * (1 - pSquareSum)) / (numberOfSamples - 1);
+    }
+
+    /**
+     * Computes Tajima's &pi; for a single site using allele counts.
+     *
+     * <p>Note: number of samples is computed using all allele counts.
+     *
+     * @param alleleCounts counts for each allele.
+     *
+     * @see #tajimasPi(int, List)
+     */
+    public static double tajimasPi(final List<Integer> alleleCounts) {
+        Verify.nonEmpty(alleleCounts, () -> "allele counts");
+        final int numberOfSamples = alleleCounts.stream().reduce(0, Integer::sum);
+        final List<Double> freqs = alleleCounts.stream()
+                .mapToDouble(c -> (double) c / (double) numberOfSamples)
+                .boxed().collect(Collectors.toList());
```

Comment:
Is it really necessary to do `mapToDouble` and then `boxed`? Maybe will be better to directly use `map`...